BoostRunClub/ViewModels/ActivityViewModel.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  ActivityViewModel.swift
+//  BoostRunClub
+//
+//  Created by 김신우 on 2020/12/07.
+//
+
+import Foundation
+
+import Combine
+
+protocol ActivityViewModelTypes: AnyObject {
+    var inputs: ActivityViewModelInputs { get }
+    var outputs: ActivityViewModelOutputs { get }
+}
+
+protocol ActivityViewModelInputs {
+    func didFilterChanged(to idx: Int)
+    func didFilterRangeChanged(range: DateRange)
+    func didSelectActivity(at index: Int)
+    func didTapShowDateFilter()
+    func didTapShowAllActivities()
+    func didTapShowProfileButton()
+
+    // Life Cycle
+    func viewDidLoad()
```

Comment:
이런 LifeCycle 과 같은 넓은 범위의 함수는 하나의 프로토콜을 만들어 이를 채택하는 형식으로 진행하는 것이 더 좋을것 같습니다. 새로운 프로토콜을 만드는 경우 혹여나 나중에 함수의 이름이 변경된다던지 혹은 새로운 LifeCycle을 추가한 경우에 쉽게 처리할 수 있는 점이 장점이라고 생각합니다.

```suggestion
protocol ViewModelLifeCycleInputsProtocol {
    func viewDidLoad()
    // viewWillAppear ...
    // deinit ...
}

protocol ActivityViewModelInputs: ViewModelLifeCycleInputsProtocol {
    func didFilterChanged(to idx: Int)
    func didFilterRangeChanged(range: DateRange)
    func didSelectActivity(at index: Int)
    func didTapShowDateFilter()
    func didTapShowAllActivities()
    func didTapShowProfileButton()
}
```